convert/main.lua

```diff
@@ -1,6 +1,15 @@
 ---@type table?
 Config.DoorList = {}
 
+local function flattenTableToArray(table)
+	local array = {}
+	for k in pairs(table) do
+		array[#array + 1] = k
+	end
+
+	return array
+end
+
```

Comment:
Wouldn't we have to convert the UI's array input to a lua set then? Seems like it would make the code complex.

Reply:
There is (or was?) already conversions like this happening for other values.
You're seemingly already converting a set-like structure into an array.

Reply:
I'm having a hard time understanding your vision here. So we'd have a table where the key can either be of type string or number and the value would be a boolean. Then before passing data to NUI, we'd convert this to an array of type string | integer or would we convert it to an array of type string, or would we attempt to maintain the table structure in NUI? Then since user input is a string, would we convert the data coming out of NUI to a number within NUI, or within Lua? Judging by your below comment, it seems like the latter.

Reply:
I'm confused by your confusion. When sending data between client and NUI we already need to convert the formats.

Client -> NUI
```ts
export const convertData = (data: DoorColumn) => {
  let newGroupsData: { name: string; grade: number }[] = [];
  if (data.groups) {
    const doorGroupsData = Object.entries(data.groups);
    for (let i = 0; i < doorGroupsData.length; i++) {
      const groupObj = doorGroupsData[i];
      newGroupsData[i] = { name: groupObj[0], grade: groupObj[1] };
    }
  }
  return { ...data, groups: [...newGroupsData], items: data.items || [] } as StoreState;
};
```
NUI -> Client
```tsx
const groupsObj: { [key: string]: number } = {};

for (let i = 0; i < data.groups.length; i++) {
  const groupField = data.groups[i];
  if (groupField.name && groupField.name !== '') groupsObj[groupField.name] = groupField.grade || 0;
}
```

For both data storage and access, I want the character ids to be stored as a set `{ [id] = 1, [id2] = 1, ... }` rather than an array. The key should either be an int or string; should be as simple as `[parseInt(id) || id] = 1`.

@LukeWasTakenn may be able to explain that better or just make the preferred change.